test/api/models/trial.js

```diff
@@ -300,6 +301,13 @@ describe('Trial', () => {
           .then((trialId) => new Trial({ id: trialId }).fetch({ withRelated: ['documents', 'documents.source'] }))
           .then((trial) => should(trial.toJSON().sources).deepEqual({}))
       ));
+
+      it('returns its own source', () => (
+        factory.create('source', { id: 'nct' })
+          .then(() => factory.create('trialWithRelated', { source_id: 'nct' }))
+          .then((trial) => (should(trial.toJSON().sources).have.key(trial.attributes.source_id))
+          )
+      ));
```

Comment:
It's also missing a few tests. For instance, what happens if you do:

```javascript
factory.create('trial')
  .then((trial) => new Trial({ id: trial.attributes.id }).fetch())
  .then((trial) => should(trial.toJSON()).be.undefined())
```

As in the trial's source wasn't loaded. Would it raise an exception? It's a bit tricky to know what `trial.is_registered` should be in this case. I considered `undefined`, as we can't say if it's registered or not because we don't have information on its source.